src/utils.py

```diff
@@ -3,65 +3,48 @@
 
 """Set of utils functions used in charm."""
 
+import logging
 import shutil
 import subprocess
 from typing import Dict, List, Optional
 
-import apt  # type: ignore[import]
 import netifaces  # type: ignore[import]
 from netaddr import IPAddress, IPNetwork  # type: ignore[import]
 from netaddr.core import AddrFormatError  # type: ignore[import]
 
+logger = logging.getLogger(__name__)
+
 
 def service_active(service_name: str) -> bool:
     """Returns whether a given service is active."""
-    result = subprocess.run(
-        ["systemctl", "is-active", service_name],
-        stdout=subprocess.PIPE,
-        encoding="utf-8",
-    )
-    return result.stdout == "active\n"
-
-
-def all_values_set(dictionary: Dict[str, str]) -> bool:
-    """Returns whether all values in a dict are set."""
-    return not any(v is None for v in dictionary.values())
+    response = shell(f"systemctl is-active {service_name}")
+    return response == "active\n"
 
 
-def install_apt(packages: List[str], update: bool = False) -> None:
-    """Installs a given list of packages."""
-    cache = apt.cache.Cache()
-    if update:
-        cache.update()
-    cache.open()
-    for package in packages:
-        pkg = cache[package]
-        if not pkg.is_installed:
-            pkg.mark_install()
-    cache.commit()
+def install_apt_packages(package_list: List[str]) -> None:
+    """Install apt package ca-certificates package."""
+    package_list_str = " ".join(package_list)
+    shell("sudo apt -qq update")
+    shell(f"sudo apt -y install {package_list_str}")
```

Comment:
I think we could use some error handling.

Reply:
There is, the `shell` method looks for the status code